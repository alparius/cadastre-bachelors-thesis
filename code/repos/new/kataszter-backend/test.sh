#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
