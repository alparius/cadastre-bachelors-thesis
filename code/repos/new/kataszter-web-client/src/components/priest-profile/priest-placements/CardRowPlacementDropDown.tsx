import React from "react";
import { Dropdown, Form, Loader, Message } from "semantic-ui-react";

import { FormattedMessage as FM, useIntl } from "react-intl";
import IParishMin from "../../../data/parish/ParishMin";
import "../../../static/priestProfile.css";
import BoldFM from "../../../util/styledMessages/BoldFM";
import ItalicFM from "../../../util/styledMessages/ItalicFM";

interface ICardRowPlacementDropDownProps {
    index: number;
    placeID: number;
    placeName: string;
    editMode: boolean;
    parishResponse: IParishMin[] | undefined;
    parishLoading: boolean;
    parishShowError: boolean;
    handlePlacementChange(placeID: number, place: string, index: number): any;
}

const CardRowPlacementDropDown: React.FC<ICardRowPlacementDropDownProps> = (props: ICardRowPlacementDropDownProps) => {
    const { placeID, placeName, editMode, handlePlacementChange, index, parishResponse, parishLoading, parishShowError } = props;
    const { formatMessage: fm } = useIntl();

    const options: any = [];

    const handleChange = (_: any, { value }: any) => {
        handlePlacementChange(options[value].key, options[value].text, index);
    };

    if (parishLoading) {
        return (
            <Form>
                <Form.Field inline>
                    <BoldFM id="profile.parish" showColon />{" "}
                    {editMode ? <Loader active inline /> : placeName ? placeName : <ItalicFM id="general.noData" />}
                </Form.Field>
            </Form>
        );
    } else if (parishShowError || !parishResponse || parishResponse.length === 0) {
        return (
            <Form>
                <Form.Field inline>
                    <BoldFM id="profile.parish" showColon />{" "}
                    {editMode ? (
                        <Message negative>
                            <FM id="error.connection" />
                        </Message>
                    ) : placeName ? (
                        placeName
                    ) : (
                        <ItalicFM id="general.noData" />
                    )}
                </Form.Field>
            </Form>
        );
    } else {
        parishResponse.forEach((resp, ndx) => {
            options.push({ key: resp.ID, text: resp.name, value: ndx });
        });

        const currOption = options.find((opt: any) => opt.key === placeID);
        const currValue = currOption ? currOption.value : null;

        return (
            <Form>
                <Form.Field inline>
                    <BoldFM id="profile.parish" showColon />{" "}
                    {editMode ? (
                        <Dropdown
                            options={options}
                            value={currValue}
                            onChange={handleChange}
                            placeholder={fm({ id: "search.select" })}
                            noResultsMessage={fm({ id: "search.noResultMini" })}
                            deburr
                            fluid
                            scrolling
                            search
                            selection
                        />
                    ) : placeName ? (
                        placeName
                    ) : (
                        <ItalicFM id="general.noData" />
                    )}
                </Form.Field>
            </Form>
        );
    }
};

export default CardRowPlacementDropDown;
