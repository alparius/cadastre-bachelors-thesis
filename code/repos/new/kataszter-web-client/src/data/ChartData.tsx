interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
