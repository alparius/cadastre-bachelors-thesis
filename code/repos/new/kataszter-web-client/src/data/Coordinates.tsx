interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
