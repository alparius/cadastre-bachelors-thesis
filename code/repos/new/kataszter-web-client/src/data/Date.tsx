interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
