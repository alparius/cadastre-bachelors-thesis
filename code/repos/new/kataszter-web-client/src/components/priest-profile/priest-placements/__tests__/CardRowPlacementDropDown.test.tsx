/*===================
  CardRowPlacementDropDown.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestParishMin } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../../util/test/test-wrapper";
import CardRowPlacementDropDown from "../CardRowPlacementDropDown";

const handlers = {
    index: 0,
    placeID: 0,
    placeName: "test",
    handlePlacementChange: jest.fn()
};

describe("CardRowPlacementDropDown element", () => {
    it("should render a dropdown list if there is no error and data is available", () => {
        const container = createComponentWithIntlAndRouter(
            <CardRowPlacementDropDown editMode parishLoading={false} parishShowError={false} parishResponse={[NewTestParishMin]} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a label if the data is loading", () => {
        const container = createComponentWithIntlAndRouter(
            <CardRowPlacementDropDown editMode parishLoading parishShowError={false} parishResponse={[NewTestParishMin]} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a label if there is a (connection) error", () => {
        const container = createComponentWithIntlAndRouter(
            <CardRowPlacementDropDown editMode parishLoading={false} parishShowError parishResponse={[NewTestParishMin]} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a label if the response is empty", () => {
        const container = createComponentWithIntlAndRouter(
            <CardRowPlacementDropDown editMode parishLoading={false} parishShowError={false} parishResponse={[]} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a label if its not in editmode", () => {
        const container = createComponentWithIntlAndRouter(
            <CardRowPlacementDropDown
                editMode={false}
                parishLoading={false}
                parishShowError={false}
                parishResponse={[NewTestParishMin]}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });
});
