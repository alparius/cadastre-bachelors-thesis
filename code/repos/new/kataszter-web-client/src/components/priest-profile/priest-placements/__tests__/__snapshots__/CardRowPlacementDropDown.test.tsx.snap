// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRowPlacementDropDown element should render a dropdown list if there is no error and data is available 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field"
  >
    <b>
      Parish
      :
    </b>
     
    <div
      aria-expanded={false}
      className="ui fluid search selection scrolling dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
    >
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value=""
      />
      <div
        aria-atomic={true}
        aria-live="polite"
        className="default text"
        role="alert"
      >
        Select from the list
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
        role="listbox"
      >
        <div
          aria-checked={false}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            name
          </span>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`CardRowPlacementDropDown element should render a label if its not in editmode 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field"
  >
    <b>
      Parish
      :
    </b>
     
    test
  </div>
</form>
`;

exports[`CardRowPlacementDropDown element should render a label if the data is loading 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field"
  >
    <b>
      Parish
      :
    </b>
     
    <div
      className="ui active inline loader"
    />
  </div>
</form>
`;

exports[`CardRowPlacementDropDown element should render a label if the response is empty 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field"
  >
    <b>
      Parish
      :
    </b>
     
    <div
      className="ui negative message"
    >
      Could not estabilish server connection.
    </div>
  </div>
</form>
`;

exports[`CardRowPlacementDropDown element should render a label if there is a (connection) error 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field"
  >
    <b>
      Parish
      :
    </b>
     
    <div
      className="ui negative message"
    >
      Could not estabilish server connection.
    </div>
  </div>
</form>
`;
